server/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java

```diff
@@ -1473,4 +1480,17 @@ protected void write(XContentBuilder b, Object value) throws IOException {
             )
         );
     }
+
+    private static KeywordFieldMapper getKeywordFieldMapperForSyntheticSource(Iterable<? extends Mapper> multiFields) {
+        for (Mapper sub : multiFields) {
+            if (sub.typeName().equals(KeywordFieldMapper.CONTENT_TYPE)) {
+                KeywordFieldMapper kwd = (KeywordFieldMapper) sub;
+                if (kwd.hasNormalizer() == false && (kwd.fieldType().hasDocValues() || kwd.fieldType().isStored())) {
```

Comment:
I think we should also check for whether ignore above is used. Otherwise there may be data loss. (as values get trimmed)

Reply:
Looks like ignore above is supported https://github.com/elastic/elasticsearch/pull/89466

Reply:
Ah, that is true. I somehow thought that it wasn't. Looking at the keyword field, if the value's length is above the configured ignore malformed then the value is also stored as a stored field (`KeywordFieldMapper#indexValue(...)` method).